pullman/storageproviders/gcs/provider_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2021 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package gcsprovider
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/kserve/modelmesh-runtime-adapter/pullman"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func newGCSRepositoryClientWithMock(t *testing.T) (*gcsRepositoryClient, *MockgcsDownloader) {
+	mockCtrl := gomock.NewController(t)
+
+	mdf := NewMockgcsDownloader(mockCtrl)
+
+	log := zap.New()
+
+	gcsrc := gcsRepositoryClient{
+		gcsclient: mdf,
+		log:       log,
+	}
+
+	return &gcsrc, mdf
+}
+
+func Test_NewRepositoryWithCredentials(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+
+	mdf := NewMockgcsDownloaderFactory(mockCtrl)
+
+	g := gcsProvider{
+		gcsDownloaderFactory: mdf,
+	}
```

Comment:
Suggestion: instead of including this code to create the mocks that is copied in the two `Test_NewRepository*` tests, could use an approach similar to the HTTP provider with a `newTestMocks` function that returns all test/mock instances used in the tests
https://github.com/kserve/modelmesh-runtime-adapter/blob/main/pullman/storageproviders/http/provider_test.go#L33-L50